neonvm/controllers/virtualmachine_controller.go

```diff
@@ -955,6 +989,43 @@ func (r *VirtualMachineReconciler) podForVirtualMachine(
 	return pod, nil
 }
 
+func (r *VirtualMachineReconciler) sshSecretForVirtualMachine(virtualmachine *vmv1.VirtualMachine) (*corev1.Secret, error) {
+	secret, err := sshSecretSpec(virtualmachine)
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the ownerRef for the Secret
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/
+	if err := ctrl.SetControllerReference(virtualmachine, secret, r.Scheme); err != nil {
+		return nil, err
+	}
+	return secret, nil
+}
+
+func sshSecretSpec(virtualmachine *vmv1.VirtualMachine) (*corev1.Secret, error) {
+	// using ed25519 signatures it takes ~16us to finish
+	publicKey, privateKey, err := sshKeygen()
+	if err != nil {
+		return nil, err
+	}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      virtualmachine.Name,
+			Namespace: virtualmachine.Namespace,
+		},
+		Immutable: ptr(true),
+		Type:      corev1.SecretTypeSSHAuth,
+		Data: map[string][]byte{
+			"ssh-publickey":  publicKey,
```

Comment:
you sometimes use `ssh-publickey` and sometimes `ssh-public-key`. Maybe it should be universalized and defined as a constant, so that we don't mix it up?

Reply:
good catch. resolved in 584358c0fa8835e68bc81858d78efd9bfcc2cd48

Reply:
Why didn't you define a constant for those? Doing this prevents such inconsistencies.